chat-commands.js

```diff
@@ -687,7 +687,7 @@ exports.commands = {
 		this.errorReply("NOTE: You probably want to use `/deleteroom` now that it exists.");
 		let id = toId(target);
 		if (!id) return this.parse('/help deregisterchatroom');
-		let targetRoom = Rooms.search(id);
+		let targetRoom = Rooms.search(id) ? Rooms.search(id).id : false;
```

Comment:
This doesn't do what you intend it to (<code>The room 'undefined' isn't registered.</code>). Line 693 is the one you need to change.